Makefile

```
@@ -95,14 +95,13 @@ $(VMLINUXH): $(OUTPUT)
 		echo "ERROR: kernel does not seem to support BTF"; \
 		exit 1; \
 	fi
-	@if [ ! -f $(VMLINUXH) ]; then \
-		if [ ! $(BPFTOOL) ]; then \
-			echo "ERROR: could not find bpftool"; \
-			exit 1; \
-		fi; \
-		echo "INFO: generating $(VMLINUXH) from $(BTFFILE)"; \
-		$(BPFTOOL) btf dump file $(BTFFILE) format c > $(VMLINUXH); \
-	fi
+	@if [ ! $(BPFTOOL) ]; then \
+		echo "ERROR: could not find bpftool"; \
+		exit 1; \
+	fi;
+
+	@echo "INFO: generating $(VMLINUXH) from $(BTFFILE)"; \
+		$(BPFTOOL) btf dump file $(BTFFILE) format c > $(VMLINUXH) || rm -f $(VMLINUXH) && exit 1
```

```suggestion
	@echo "INFO: generating $(VMLINUXH) from $(BTFFILE)"; \
	$(BPFTOOL) btf dump file $(BTFFILE) format c > $(VMLINUXH) || rm -f $(VMLINUXH) && exit 1
```

Can we also add a log saying failed to create vmlinux.h file before exiting?

---

BTW, instead of obligating the end user to have bpftool installed (which isn't needed in case of tracee, for example), we could include a "recent enough" vmlinux.h file and rely in libbpf CO-RE functionality to do the relocations.

---

> Can we also add a log saying failed to create vmlinux.h file before exiting?

Ok.

> BTW, instead of obligating the end user to have bpftool installed (which isn't needed in case of tracee, for example), we could include a "recent enough" vmlinux.h file and rely in libbpf CO-RE functionality to do the relocations.

It might work. We need to do some tests. BTW, this PR is a temporary workaround once we'll focus on https://github.com/aquasecurity/libbpfgo/issues/177 later.